subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/FileZipInput.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.changedetection.state;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.Enumeration;
+import java.util.function.Supplier;
+import java.util.zip.ZipFile;
+
+class FileZipInput extends ZipInput {
+
+    private final ZipFile file;
+    private final Enumeration<? extends java.util.zip.ZipEntry> entries;
+
+    public FileZipInput(File file) throws IOException {
```

Comment:
Constructors that throw checked exception are sometimes painful to handle.
If you think it would make the usage cleaner, consider replacing them with rethrown `UncheckedException.throwAsUncheckedException`

Reply:
Done